src/v/serde/parquet/encoding.h

```diff
@@ -21,20 +21,113 @@ namespace serde::parquet {
 //
 // See:
 // https://parquet.apache.org/docs/file-format/data-pages/encodings/#plain-plain--0
+template<typename value_type>
+class plain_encoder;
 
-iobuf encode_plain(const chunked_vector<boolean_value>& vals);
+template<>
+class plain_encoder<boolean_value> {
```

Comment:
nit: can we move the implementation of the specialization into the cc file? I've been trying to keep stuff out if headers, even if there is only a single reference to it today, it's likely going to expand over time.

Reply:
Yeah, that's easily done. Will do so if we decide to go with the design in this PR.